src/core.ts

```diff
@@ -65,36 +66,41 @@ export function activate() {
     onActivateError(e);
     return;
   }
+
+  state = State.Activated;
 }
 
 export function teardown() {
-  for (let plugin of activePlugins) {
-    plugin.teardown();
-  }
+  if (state === State.Activating || state === State.Activated) {
```

Comment:
>Activating [](start = 22, length = 10)

Is it OK to do this during the "Activating" state? Why? Would be good to add a comment to explain this more.

Reply:
We can encounter errors during init phase, in which case we want to clean up.

---
In reply to: [158272666](https://github.com/Microsoft/clarity-js/pull/45#discussion_r158272666) [](ancestors = 158272666)